scalafmt-tests/src/test/resources/default/TypeWith.stat

```diff
@@ -9,3 +9,13 @@
       annotationOwnerLookUp: ScLiteral => Option[
           PsiAnnotationOwner with PsiElement]): Option[PsiAnnotationOwner] = 1
 }
+<<< #1133
+implicit val catsStdInstancesForOption
+   : Traverse[Option] with MonadError[Option, Unit] with Alternative[Option]
+       with CommutativeMonad[Option] with CoflatMap[Option] = ???
+>>>
+implicit val catsStdInstancesForOption: Traverse[Option]
```

Comment:
This may be more ambitious that you were planning for (so "no" is an acceptable answer), but while were fixing with chains, could we add a test case for?
```scala
new A with B with C with D ...
```
I've seen surprising behavior in these scenarios and I suspect fixing it likewise is a matter of adding another case to `WithChain`

Reply:
Yes no problem (I think)

Reply:
I might need some help with this. I don't think this falls into the `Type.With` case and instead hits the `Template` case.

My thought is to treat this like the `KwExtends` case above and do

```patch
       case tok @ FormatToken(_, right @ KwWith(), _) =>
         rightOwner match {
+          // Anonymous template (new A with B { ... })
+          case template: Template if template.parent.exists(_.is[Term.New]) =>
+            val lastToken = template.tokens.last
+            binPackParentConstructorSplits(
+              Set(template),
+              lastToken,
+              2
+            )
           case template: Template =>
             val hasSelfAnnotation = template.self.tokens.nonEmpty
```

This almost works but it indents after every with

```
new A
  with B
    with C
      with D
```

Any ideas?

Reply:
I pushed the non-working change mentioned above. I'm assuming the fix is something analogous to the `isFirstWith` logic but I'm not sure how to achieve it with the template

Reply:
Hmm, the changes seem unrelated enough to merit a separate PR. Your change already improves on the state-of-the-art so I propose you drop the last commit, we merge and move this case to another ticket.

I'll need to locally play around with this case to see how would be best to solve it.

Reply:
SGTM 👍. I rolled it back